src/bors/handlers/refresh.rs

```diff
@@ -28,6 +28,36 @@ pub async fn refresh_repository(
     }
 }
 
+pub async fn sync_prs_state(repo: Arc<RepositoryState>, db: Arc<PgDbClient>) -> anyhow::Result<()> {
+    let repo = repo.as_ref();
+    let db = db.as_ref();
+    let repo_name = repo.repository();
+    // load prs from github
+    let prs = repo.client.fetch_prs().await?;
+    let pr_to_status = db.get_all_pull_request_status(repo_name).await?;
+
+    for pr in prs {
```

Comment:
As noted in my other comment, and in the original issue, we cannot simply download all PRs from GitHub, there's too many. We only have to fetch the open ones and synchronize smartly based on that.